File: CPP/Assignment/day7/seperateEmployee/Employee.cpp
#include "Employee.h"

void Employee::Accept() {
    cout << "Please enter empNo:->";
    cin >> empNo;
    cout << "Please enter employee name:->";
    cin >> name;
    cout << "Please enter employee salary:->";
    cin >> salary;
    cout << "Please enter employee dept:->";
    cin >> dept;
}

 
void Employee::Display() {
    cout << "Employee empNo:->" << empNo << endl;
    cout << "Employee name:->" << name << endl;
    cout << "Employee salary:->" << salary << endl;
    cout << "Employee dept:->" << dept << endl;
}

 
string Employee::getEmpNo() const {
    return empNo;
}

 
string Employee::getName() const {
    return name;
}

 
double Employee::getSalary() const {
    return salary;
}

 
bool Employee::compare(const Employee &e1, const Employee &e2) {
    if (e1.getEmpNo() != e2.getEmpNo()) {
        return e1.getEmpNo() < e2.getEmpNo();
    } else if (e1.getName() != e2.getName()) {
        return e1.getName() < e2.getName();
    }
    return e1.getSalary() < e2.getSalary();
}

File: OOPWJAVA/SessionData/Day6/EmployeeExample/src/com/cdac/acts/Employee.java
package com.cdac.acts;

public class Employee {
	private int empId;
	private String name;
	private String dept;
	private double salary;
	
	private static int idGenerator = 1000;
	
	public Employee(){
		this.empId = idGenerator++;
		this.name = "";
		this.dept = "";
		this.salary = 0.0;
	}
	
	public Employee(String name, String dept, double salary){
		this.empId = idGenerator++;
		this.name = name;
		this.dept = dept;
		this.salary = salary;
	}
	
	public int getEmpId() {
		return empId;
	}
	
	@Override
	public String toString() {
		return "Employee[empId=" + empId 
				+ " name=" + name + " dept=" + dept
				+ " salary=" + salary + "]";
	}
	
}
